wandb/sdk/lib/filesystem.py

```diff
@@ -13,6 +13,9 @@ def mkdir_exists_ok(dir_name: AnyPath) -> None:
         FileExistsError: if `dir_name` exists and is not a directory.
         PermissionError: if `dir_name` is not writable.
     """
+    if not os.access(dir_name, os.W_OK):
+        raise PermissionError(f"{dir_name!s} is not writable")
+
     os.makedirs(dir_name, exist_ok=True)
     if not os.access(dir_name, os.W_OK):
         raise PermissionError(f"{dir_name!s} is not writable")
```

Comment:
This feels like a better way of writing this, if we're trying to offer more informative error messages:
```suggestion
    try:
        os.makedirs(dir_name, exist_ok=True)
    except FileExistsError as e:
        raise FileExistsError(f"{dir_name!s} exists and is not a directory") from e
    except PermissionError as e:
        raise PermissionError(f"{dir_name!s} is not writable") from e
```